macro-kroki-api/src/checkstyle/checkstyle-suppressions.xml

```diff
@@ -25,4 +25,5 @@
   "http://www.puppycrawl.com/dtds/suppressions_1_0.dtd">
 
 <suppressions>
+  <suppress files="." checks="ParameterNumberCheck"/>
```

Comment:
You're excluding too many files. This means we won't notice when someone adds too many parameters from now on (which is a sign of bad design). In this particular case, if we want to keep ``KrokiContainerConfiguration`` read-only (which is probably OK) then using the builder pattern can help reduce the number of constructor parameters. See for instance https://www.digitalocean.com/community/tutorials/builder-design-pattern-in-java (something I found on a quick search, there may be other / better examples, but this one is good enough).

Reply:
I have now pushed my suggestion for a builder pattern implementation. I have tested it to be functional on my XWiki installation.